app/_how-tos/operator-controlplane-config-dump.md

```diff
@@ -0,0 +1,304 @@
+    diagnostic endpoints to inspect configurations for debugging and troubleshooting.
+---
+
+The dump config server is a diagnostic feature in {{ site.operator_product_name }} that allows you to inspect the Kong configuration generated for each ControlPlane. This is particularly useful when debugging configuration issues or understanding how your Kubernetes resources are translated to Kong configuration.
+
+## Enable the dump config server
+
+The dump config server is disabled by default. You must enable it when installing or upgrading {{ site.operator_product_name }}.
+
+### Using Helm
+
+1. Enable the dump config server by setting the values during installation:
+
+   ```bash
+   helm upgrade --install kong-operator kong/kong-operator \
+     --set enableControlplaneConfigDump=true \
+     --set controlplaneConfigDumpPort=10256 \
+     --namespace kong-system \
+     --create-namespace
+   ```
+
+   Alternatively, create a values file:
+
+   ```yaml
+   # operator-values.yaml
+   enableControlplaneConfigDump: true
+   controlplaneConfigDumpPort: 10256  # Optional: defaults to 10256
+   ```
+
+   Then install with:
+
+   ```bash
+   helm upgrade --install kong-operator kong/kong-operator \
+     -f operator-values.yaml \
+     --namespace kong-system \
+     --create-namespace
+   ```
+
+3. Verify the dump config service is created:
+
+   ```bash
+   kubectl get service -n kong-system kong-operator-controlplane-config-dump
+   ```
+
+   The output should show:
+   ```
+   NAME                                     TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)     AGE
+   kong-operator-controlplane-config-dump   ClusterIP   10.96.100.150   <none>        10256/TCP   1m
+   ```
+
+## Access the dump config server
+
+The dump config server is not exposed externally by default. Use port-forwarding to access it locally:
+
+```bash
+kubectl port-forward -n kong-system service/kong-operator-controlplane-config-dump 10256:10256
+```
+
+Keep this terminal open while you work with the dump config server.
+
+## List managed ControlPlanes
+
+To see all ControlPlanes managed by the operator:
+
+```bash
+curl http://localhost:10256/debug/controlplanes | jq .
+```
+
+Example response:
+```json
+{
+  "controlPlanes": [
+    {
+      "namespace": "default",
+      "name": "my-controlplane",
+      "id": "a1b2c3d4-e5f6-7890-abcd-ef1234567890"
+    },
+    {
+      "namespace": "production",
+      "name": "prod-cp",
+      "id": "b2c3d4e5-f6a7-8901-bcde-f23456789012"
+    }
+  ]
+}
+```
+
+### Filter by namespace
+
+To list ControlPlanes in a specific namespace:
+
+```bash
+curl http://localhost:10256/debug/controlplanes/namespace/production | jq .
+```
+
+## Dump ControlPlane configuration
+
+To inspect the Kong configuration for a specific ControlPlane:
+
+```bash
+# Replace with your ControlPlane's namespace and name
+curl http://localhost:10256/debug/controlplanes/namespace/default/name/my-controlplane/config/
```

Comment:
```suggestion
curl http://localhost:10256/debug/controlplanes/namespace/default/name/my-controlplane/config/successful
```
This API returns the configuration in the latest successful application to Kong gateway.